orangecontrib/prototypes/widgets/oworaclesql.py

```diff
@@ -0,0 +1,136 @@
+from AnyQt.QtWidgets import QSizePolicy, QPlainTextEdit, QHBoxLayout, QLineEdit
+from Orange.widgets import gui, settings
+from Orange.widgets.widget import OWWidget, Msg, Output
+from Orange.data import Table, Domain, ContinuousVariable, DiscreteVariable, TimeVariable
+import sip
+import datetime
+try:
+    import cx_Oracle
+except:
+    cx_Oracle = None
+    
+
+        
+class ORACLESQL(OWWidget):
+
+    name = "Oracle SQL"
+    icon = "icons/OracleSQL.svg"
+    want_main_area = True
+    want_message_bar = True
+    #inputs = []
+    outputs = [("Data", Table)]
+    description = "Select data from oracle databases"
+    settingsHandler = settings.DomainContextHandler()
+    autocommit = settings.Setting(False, schema_only=True)
+    savedQuery = settings.Setting(None, schema_only=True)
+    savedUsername = settings.Setting(None, schema_only=True)
+    savedPwd = settings.Setting(None, schema_only=True)
+    savedDB = settings.Setting(None, schema_only=True)
+    
+    class Error(OWWidget.Error):
+        no_backends = Msg("Please install cx_Oracle package. It is either missing or not working properly")
+   
+    class Outputs:
+        data = Output("Data", Table)       
+
+    def __init__(self):
+        super().__init__()
+
+        #Defaults
+        self.setSizePolicy(QSizePolicy.Maximum, QSizePolicy.Maximum)
+        self.domain = None
+        self.data = None
+        self.query = ''
+        if self.savedQuery is not None:
+            self.query = self.savedQuery
+        self.username = ''
+        if self.savedUsername is not None:
+            self.username = self.savedUsername
+        self.password = ''
+        if self.savedPwd is not None:
+            self.password = self.savedPwd
+        self.database = ''
+        if self.savedDB is not None:
+            self.database = self.savedDB
+        #Control Area layout
+        sip.delete(self.controlArea.layout())
```

Comment:
I would just set the new layout and leave deleting of the old one to the gc.

Reply:
I'm deleting the original layout because the line  self.controlArea.setLayout(QHBoxLayout()) is not working without it - The SQL box is being placed under the Database box. 
If there is some other way to place the SQL box to the right of the Database box - i can change.

Reply:
Have you considered placing the sql box into the main area instead?